sdv/data_processing/data_processor.py

```diff
@@ -346,115 +346,234 @@ def _get_transformer_instance(self, sdtype, column_metadata):
 
         return deepcopy(transformer)
 
-    def _create_config(self, data):
-        sdtypes = {}
-        transformers = {}
+    def _handle_multi_column(self, column, data):
+        """Handle columns that are part of multi-column transformers.
 
-        columns_in_multi_col_transformer = self._get_grouped_columns()
-        for column in set(data.columns):
-            column_metadata = self.metadata.columns.get(column)
-            sdtype = column_metadata.get('sdtype')
+        Args:
+            column (str):
+                Column name.
+            data (pandas.DataFrame):
+                The input data.
+
+        Returns:
+            tuple:
+                (sdtype, transformer) where transformer is None for multi-column cases.
+        """
+        column_metadata = self.metadata.columns.get(column)
+        sdtype = column_metadata.get('sdtype')
+
+        return sdtype, None
 
-            if column in columns_in_multi_col_transformer:
-                sdtypes[column] = sdtype
-                continue
+    def _handle_id_column(self, column, data):
+        """Handle columns with 'id' sdtype.
 
-            pii = (
-                sdtype not in self._transformers_by_sdtype
-                and sdtype not in {'unknown', 'id'}
-                and (column_metadata.get('pii', True))
+        Args:
+            column (str):
+                Column name.
+            data (pandas.DataFrame):
+                The input data.
+
+        Returns:
+            tuple:
+                (sdtype, transformer)
+        """
+        column_metadata = self.metadata.columns.get(column)
+        sdtype = column_metadata.get('sdtype')
```

Comment:
This is repeated in every `handle` function, maybe we can pass it from the `_get_column_config`.